resources/todomvc/architecture-examples/svelte/src/Item.svelte

```diff
@@ -0,0 +1,55 @@
+<script>
+    import { createEventDispatcher, tick } from 'svelte';
+
+    let editing = false;
+
+    const dispatch = createEventDispatcher();
+
+    function removeItem() {
+        dispatch('removeItem');
+    }
+
+    function startEdit() {
+       editing = true;
+    }
+
+    function handleEdit(event) {
+        if (event.key === "Enter")
+            event.target.blur();
+        else if (event.key === "Escape") 
+            editing = false;
+    }
+
+    function updateItem(event) {
+        if (!editing) return;
+        dispatch('updateItem', { text: event.target.value });
+        editing = false;
+    }
```

Comment:
I think that the update operation could be done directly here. If the value is emtpy we could call `removeItem`.